.github/workflows/test.yml

```diff
@@ -53,28 +73,168 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 20
+          check-latest: true
+          cache: "npm"
+          cache-dependency-path: "**/package.json"
+      - name: Cache npm dependencies
+        uses: actions/cache@v3
+        with:
+          path: |
+            ./runtime/node_modules
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-npm-
+      - name: Install script dependencies
+        run: npm install --prefix ./.ci
       - name: Setup dotnet
         uses: actions/setup-dotnet@v4
         with:
           dotnet-version: "8.0.x"
+      - name: Cache NuGet packages
+        uses: actions/cache@v3
+        with:
+          path: ~/.nuget/packages
+          key: ${{ runner.os }}-nuget-${{ hashFiles('**/*.csproj') }}
+          restore-keys: ${{ runner.os }}-nuget-
       - name: Run tests
         run: ./.ci/test.sh
         env:
           DOTNET_CONSOLE_ANSI_COLOR: true
+          
+  container-test-builds:
+    name: Build ${{matrix.tag}} Container
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - dockerfile: ConsumerApi/Dockerfile
+            tag: consumer-api
+          - dockerfile: AdminApi/src/AdminApi/Dockerfile
+            tag: admin-ui
+          - dockerfile: Modules/Devices/src/Devices.AdminCli/Dockerfile
+            tag: seed-client
+          - dockerfile: DatabaseMigrator/Dockerfile
+            tag: database-migrator
+          - dockerfile: EventHandlerService/src/EventHandlerService/Dockerfile
+            tag: event-handler-service
```

Comment:
Is there a specific reason why you only build these five images instead of all?

Also, you should be able to remove the jobs that are specifically for building the images. That's just duplication, I think.

Reply:
These are the only images containing a build instruction in the compose files (as this was my focus)

Reply:
Yeah, I was expecting that. But I like your solution for two reasons:
1. It leads to less code
2. It then also benefits from the caching of the used base images.

Reply:
Why did you ignore this comment?

Reply:
I didn't ignore it, I had to work on the release v5 PR, but thought about a solution